plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.processor;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+import static org.opensearch.ml.processor.InferenceProcessorAttributes.IGNORE_MISSING;
+import static org.opensearch.ml.processor.InferenceProcessorAttributes.INPUT_MAP;
+import static org.opensearch.ml.processor.InferenceProcessorAttributes.MODEL_CONFIG;
+import static org.opensearch.ml.processor.InferenceProcessorAttributes.MODEL_ID;
+import static org.opensearch.ml.processor.InferenceProcessorAttributes.OUTPUT_MAP;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.client.Client;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+import org.opensearch.ingest.ValueSource;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.transport.MLTaskResponse;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskAction;
+import org.opensearch.script.ScriptService;
+import org.opensearch.script.TemplateScript;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+
+/**
+ * MLInferenceIngestProcessor requires a modelId string to call model inferences
+ * maps fields in document for model input, and maps model inference output to new document fields
+ * this processor also handles dot path notation for nested object( map of array) by rewriting json path accordingly
+ */
+public class MLInferenceIngestProcessor extends AbstractProcessor implements ModelExecutor {
+
+    private final InferenceProcessorAttributes inferenceProcessorAttributes;
+    private final boolean ignoreMissing;
+    private final ScriptService scriptService;
+    private static Client client;
+    public static final String TYPE = "ml_inference";
```

Comment:
Should we use `ml_inference` or just `inference` ?

Reply:
`ml_inference` emphasizes it requires a ml model in this processor.